### PR TITLE
implement mesh dataset processing for model

### LINKED ISSUE #46920: Mesh dataset input in modeler can not be linked to a parent mesh
### What is the bug or the crash?

In modeler, we can add an mesh dataset input but we can not define a parent mesh.
So when model is started, mesh dataset selection is always empty.

image
image
image

### Steps to reproduce the issue

1. Just create a model with 2 inputs : meshlayer and mesh dataset
2. Run the model and try to select a dataset

### Versions

windows 3.22.3

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vcloarec**
fix #46920
The part of mesh dataset group parameter and dataset time parameter necessary to make working this parameters with model was simply not implemented.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Can you add tests to testqgsprocessinggui too?

File: src/gui/processing/qgsprocessingmeshdatasetwidget.cpp [hunk 1]
```diff
@@ -740,4 +750,120 @@ void QgsProcessingMeshDatasetTimeWidget::updateValue()
...
+        mParentLayerComboBox->addItem( def->description(), def->name() );
+      }
+    }
+  }
+
+  const QgsProcessingParameterMeshDatasetGroups *datasetGroupDef =
+    static_cast< const QgsProcessingParameterMeshDatasetGroups *>( definition );
+  int currentIndex = mParentLayerComboBox->findData( datasetGroupDef->meshLayerParameterName() );
+  if ( currentIndex != -1 )
+    mParentLayerComboBox->setCurrentIndex( currentIndex );
```
Comment (nyalldawson) [inline src/gui/processing/qgsprocessingmeshdatasetwidget.cpp:787]: elsewhere we do this:

```suggestion
    else if ( !datasetGroupDef->meshLayerParameterName().isEmpty() )
    {
        // if no layer parameter candidates found, we just add the existing one as a placeholder
        mParentLayerComboBox->addItem( datasetGroupDef->meshLayerParameterName(), datasetGroupDef->meshLayerParameterName() );
        mParentLayerComboBox->setCurrentIndex( mParentLayerComboBox->count() - 1 );
    }
```

File: tests/src/gui/testprocessinggui.cpp [hunk 2]
```diff
@@ -10043,6 +10043,22 @@ void TestProcessingGui::testMeshDatasetWrapperLayerInProject()
...
   QCOMPARE( pythonString, QStringLiteral( "{'type': 'defined-date-time','value': QDateTime(QDate(2020, 2, 1), QTime(3, 2, 1))}" ) );
+
+  // parameter definition widget
+  std::unique_ptr<QgsProcessingAbstractParameterDefinitionWidget> paramWidgetGroup( groupsWrapper.createParameterDefinitionWidget( context, widgetContext, &groupsDefinition, nullptr ) );
+  QgsProcessingParameterDefinition::Flags flags;
+  std::unique_ptr<QgsProcessingParameterDefinition> paramDefGroup( paramWidgetGroup->createParameter( QStringLiteral( "my_param_name" ), QStringLiteral( "my_param_descr" ), flags ) );
+  QVERIFY( paramDefGroup );
+  QCOMPARE( paramDefGroup->name(), QStringLiteral( "my_param_name" ) );
+  QCOMPARE( paramDefGroup->description(), QStringLiteral( "my_param_descr" ) );
+  QCOMPARE( paramDefGroup->type(), QgsProcessingParameterMeshDatasetGroups::typeName() );
```
Comment (nyalldawson) [inline tests/src/gui/testprocessinggui.cpp:10054]: should also be checking that the parent layer parameter selection is working here (i.e. that it survives the round trip through the dialog)

Comment (vcloarec) [reply tests/src/gui/testprocessinggui.cpp:10054]: done.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**